UCLALibrary/oral-history-staff-ui
# Sys 1052/create projectitem model

Initial PR for ProjectItem essential fields. Please review the arguments on User related foreign keys, namely

```
models.ForeignKey(
        settings.AUTH_USER_MODEL,
        on_delete=models.SET_NULL,
        blank=True,
        null=True,
        related_name="+",
    )
```
The logical behavior should be as follows:
- Blank fields are allowed
- Null values are allowed
- If the user pointed to by this field is deleted, the result should be updated to null
- related_name was added to prevent reverse relationship from User table being created, not using this caused a conflict due to 2 User references in this table (created_by, late_modified by). Details at https://docs.djangoproject.com/en/4.1/ref/models/fields/#django.db.models.ForeignKey.related_name

Testing is similar to previous models:
- Run migrations, or restart containers
- Run `$ docker-compose exec django python manage.py loaddata --app oh_staff_ui seed-data` to import data
- Confirm content via admin ui
